client/src/components/ColumnArea/ColumnArea.js

```diff
@@ -0,0 +1,100 @@
+import React, { useContext } from "react";
+import Column from "../Column/Column";
+import { makeStyles } from "@material-ui/core/styles";
+import { UserContext } from "../../userContext";
+import { DragDropContext, Droppable } from "react-beautiful-dnd";
+
+const ColumnArea = props => {
+  const classes = useStyles(props);
+  const { value1 } = useContext(UserContext);
+  let [taskState, setTaskState] = value1;
+
+  const onDragEnd = result => {
+    const { destination, source, draggableId, type } = result;
+    if (!destination) {
+      return;
+    }
+    //Check if it is dropped to same column and same index
+    if (destination.droppableId === source.droppableId && destination.index === source.index) {
+      return;
+    }
+
+    //For moving column
+    if (type === "column") {
+      const newColumnOrder = Array.from(taskState.columnOrder);
+      newColumnOrder.splice(source.index, 1);
+      newColumnOrder.splice(destination.index, 0, draggableId);
+      setTaskState({ ...taskState, columnOrder: newColumnOrder });
+      return;
+    }
+
+    const start = taskState.columns[source.droppableId];
+    const finish = taskState.columns[destination.droppableId];
+
+    //for the case card move around in same column
+    if (start === finish) {
+      const newTaskIds = Array.from(start.taskIds);
```

Comment:
code seems a little unorganized here with repeated bits and functions that could be implemented in a more efficient / less cluttered way.